File: week6/objective-c/PulseSecureVpnManager.m
static void* libPulseSecureHandle;
static void (*startProxy)();

- (void) connectProxy{
    AFWrapper *wrapper = [[AFWrapper alloc] init];
    libPulseSecureHandle = [wrapper getLibPulseSecure];

    if (libPulseSecureHandle != NULL) {
        startProxy = dlsym(libPulseSecureHandle, "startProxy");
        if (startProxy != NULL)  {
            startProxy();
            // Connected!
            // Launch kPulseSecureVpnConnectedNotification notification
            [[NSNotificationCenter defaultCenter] postNotificationName:kPulseSecureVpnConnectedNotification object:self userInfo:nil];
        }
    }
}

File: week6/javascript/PulseSecureVpnManager.js
this.handlers = [];

function PulseSecureVpnManager(handler) {
    this.handlers.push(handler);
}

this.libPulseSecureHandle = null;

function connectProxy() {
    var wrapper = new AFWrapper();
    this.libPulseSecureHandle = wrapper.getLibPulseSecure();

    if (this.libPulseSecureHandle != NULL) {
        var startProxy = dlsym(libPulseSecureHandle, "startProxy");
        startProxy();
        // Connected!
        this.handlers.forEach(function(handler) {
            handler(); //launch event
        });
    }
}
